src/ClassParser/CDClassDefinitionParser.class.st

```diff
@@ -20,9 +20,19 @@ CDClassDefinitionParser class >> fromASTNode: aNode [
 
 { #category : #parsing }
 CDClassDefinitionParser >> handleClassName: aNode withType: aSymbol [
+	| layout |
 	
 	self handleClassName: aNode.
-	classDefinition classKind: aSymbol
+	layout := FixedLayout.
+	aSymbol = #variableWordSubclass: ifTrue: [ layout := WordLayout ].
+	aSymbol = #ephemeronSubclass: ifTrue: [ layout := EphemeronLayout ].
+	aSymbol = #weakSubclass: ifTrue: [ layout := WeakLayout ].
+	aSymbol = #variableByteSubclass: ifTrue: [ layout := ByteLayout  ].
+	aSymbol = #variableSubclass: ifTrue: [ layout := VariableLayout ].
+	aSymbol = #immediateSubclass: ifTrue: [ layout := ImmediateLayout ].
+	aSymbol = #subclass: ifTrue: [ layout := FixedLayout ].
```

Comment:
It can be simplified with some lookup method based on  #subclassDefiningSymbol.

Reply:
Yes :)
I will integrate this one and we continue.

Reply:
I added https://github.com/pharo-project/pharo/issues/9961